packages/api/src/command/medical/doc-request/delete-doc-request.ts

```diff
@@ -0,0 +1,23 @@
+import { validateVersionForUpdate } from "../../../models/_default";
+import { BaseUpdateCmdWithCustomer } from "../base-update-command";
+import { getDocRequestOrFail } from "./get-doc-request";
+
+export type DocRequestDeleteCmd = BaseUpdateCmdWithCustomer;
+
+export type DeleteOptions = {
+  allEnvs?: boolean;
+};
+
+export const deleteDocRequest = async (
+  requestDelete: DocRequestDeleteCmd,
+  options: DeleteOptions = {}
+): Promise<void> => {
+  const { id, cxId, eTag } = requestDelete;
+
+  const request = await getDocRequestOrFail({ id, cxId });
+  validateVersionForUpdate(request, eTag);
+
+  if (options.allEnvs) {
+    await request.destroy();
```

Comment:
I modelled this after destroying a patient, but was not sure why the destroy would only occur if option.allEnvs was true.